app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class ExpensesController < ApplicationController
+  before_action :authorize_user
+
+  def index
+    render json: current_user.expenses, status: :ok
+  end
+
+  def show
+    return render_not_found('Expense not found') if expense.blank?
```

Comment:
Acho que para simplificar esse passo, poderiamos adicionar uma `before_action` para os metodos `show`, `update` e `destroy`, algo como `find_expense`, ou `set_expense`. Ai essa before action eh responsavel por procurar o model e, caso nao encontre, renderizar o "not found"

Reply:
Nessa parte eu discordo 😅 

Sei que o problema maior mesmo é usar callbacks do ActiveRecord, mas eu já evito qualquer um porque eu acho que causa muita indireção no código. Talvez aqui a gente não sinta muito porque o código ainda tá pequeno, mas pensando que esse código pode crescer, isso pode ser um problema no futuro, e eu inclusive já tive vários problemas com isso, tanto de eu ficar perdido na hora de entender um pedaço de código, quanto de eu ver outras pessoas pedindo ajuda porque não entende como tal coisa acontece devido a codebase estar cheia de callbacks. Então se possível, eu sempre tento pensar em uma solução diferente que não precise disso

Então eu voto +1 pra ficar do jeito que está 😄